src/Robo/Plugin/Commands/DevelopmentModeCommands.php

```diff
@@ -163,26 +165,63 @@ protected function configureAwsCredentials(string $awsConfigDirPath, string $aws
      *
      * @return array
      *   An S3 request object configuration array.
+     */
+    protected function s3BucketRequestConfig(string $siteName): array
+    {
+        $s3ConfigArray = ['Bucket' => $this->s3BucketForSite($siteName)];
+        try {
+            $s3KeyPrefix = $this->getConfig('database_s3_key_prefix_string', $siteName);
+            $this->say("'$siteName' S3 Key prefix: '$s3KeyPrefix'");
+            $s3ConfigArray['Prefix'] = $s3KeyPrefix;
+        } catch (TaskException $e) {
+            $this->say("No S3 Key prefix found for $siteName.");
+        }
+        return $s3ConfigArray;
+    }
+
+    /**
+     * Get S3 Bucket for site.
+     *
+     * @param string $siteName
+     *   The site name.
+     *
+     * @return string
+     *   An S3 bucket.
      *
      * @throws \Robo\Exception\TaskException
      */
-    protected function s3BucketRequestConfig(string $siteName): array
+    protected function s3BucketForSite(string $siteName): string
     {
         if (!$bucket = $this->getConfig('database_s3_bucket', $siteName)) {
             throw new TaskException($this, "database_s3_bucket value not set for '$siteName'.");
         }
         $this->say("'$siteName' S3 bucket: $bucket");
+        return $bucket;
+    }
 
-        $s3ConfigArray = ['Bucket' => $bucket];
+    /**
+     * Get S3 region for site.
+     *
+     * @param string $siteName
+     *   The site name.
+     *
+     * @return string
+     *   An S3 region.
+     */
+    protected function s3RegionForSite(string $siteName): string
```

Comment:
Should this leverage `s3BucketRequestConfig`?

Reply:
I don't follow.

Reply:
I was thinking that `s3BucketRequestConfig` was getting all config in a generic way, but I think that assumption was wrong. It's getting config but then building something unique.

Reply:
s3BucketRequestConfig is also config needed for a request, whereas this is needed to instantiate the client.